pilot/pkg/model/xds_cache.go

```diff
@@ -15,13 +15,47 @@
 package model
 
 import (
+	"fmt"
+	"istio.io/pkg/monitoring"
 	"sync"
 
 	"github.com/golang/protobuf/ptypes/any"
 
+	"github.com/hashicorp/golang-lru"
+
 	"istio.io/istio/pilot/pkg/util/sets"
 )
 
+var (
+	writes = monitoring.NewSum(
+		"xds_cache_writes",
```

Comment:
This is recorded 1-2x per resource. In the past, we recorded a metric on EDS resources and found it caused ~4% overhead on pilot CPU: https://github.com/istio/istio/issues/25154. This was without caching, I am a bit worried with caching it may be even greater percentage as the efficiency of the rest of the code is improved. Would be good to measure this in a live cluster if possible

Reply:
Just for write, or hit/miss as well.

Reply:
We do a lookup for every resource though. So say we have 1000 pods and 1000 clusters, each push we do 1k*1k=1 million stats writes